File: server/models/child.js
// load the things we need
var mongoose = require('mongoose');

// required to refernce to ObjectId
var ObjectId = mongoose.Schema.Types.ObjectId;

// define the schema for our child model
var childSchema = mongoose.Schema({
	name: String,
	age: Number,
	gender: String,
	sports: [String],
	diseases: [String],
	log: [{
		date: {
			type: Date,
			default: Date.now
		},
		bloodsugar: Number,
		be: Number,
		beFactor: Number,
		correctionValue: Number,
		insulin: Number,
		mood: String,
		notes: String
	}],
	parent: ObjectId,
	doc: ObjectId,
	therapy: {
		factor: {
			morning: Number,
			day: Number,
			evening: Number,
			type: {
				type: String
			}
		},
		type: {
			type: String
		},
		target: Number,
		correction: Number
	}
}, {
	collection: 'children',
	versionKey: false
});

// create the model for children and expose it to our app
module.exports = mongoose.model('Child', childSchema);

File: client/app/src/main/java/util/Child.java
package util;

import android.app.Activity;

import java.util.Date;
import java.util.List;

public class Child {
    public String name;
    public Integer age;
    public String gender;
    public List<String> sports;
    public List<String> diseases;
    public List<LogEntry> log;
    public String parent;
    public String doc;
    public Therapy therapy;

    public class Therapy {
        public Factor factor;
        public String type;
        public Integer target;
        public Integer correction;
    }

    public class Factor {
        public Double morning;
        public Double day;
        public Double evening;
        public String type;
    }

    public static class LogEntry {
        public String date;
        public Integer bloodsugar;
        public Double be;
        public Double beFactor;
        public Integer correctionValue;
        public Double insulin;
        //public String mood;
        public String notes;
    }
}